src/updatechecker.cpp

```diff
@@ -219,9 +220,34 @@ UpdateChecker::UpdateChecker(): Thread("WinSparkle updates check")
 
 void UpdateChecker::Run()
 {
-    // no initialization to do, so signal readiness immediately
-    SignalReady();
+    while (1)
+    {
+        // no initialization to do, so signal readiness immediately
+        SignalReady();
```

Comment:
This should only be called once (and the purpose, which implies this, is documented).